bokehjs/src/coffee/models/annotations/legend_item.coffee

```diff
@@ -0,0 +1,74 @@
+_ = require "underscore"
+
+Model = require "../../model"
+
+p = require "../../core/properties"
+{logger} = require "../../core/logging"
+
+class LegendItem extends Model
+  type: "LegendItem"
+
+  _check_data_sources_on_renderers: () ->
+    field = @get_field_from_label_prop()
+    if field?
+      if @renderers.length < 1
+        return false
+      source = @renderers[0].data_source
+      if source?
+        for r in @renderers
+          if r.data_source != source
+            return false
+    return true
+
+  _check_field_label_on_data_source: () ->
+    field = @get_field_from_label_prop()
+    if field?
+      if @renderers.length < 1
+        return false
+      source = @renderers[0].data_source
+      if source? and field not in source.columns()
+        return false
+    return true
+
+
+  initialize: (attrs, options) ->
+    super(attrs, options)
+    # Validate data_sources match
+    data_source_validation = @_check_data_sources_on_renderers()
+    if not data_source_validation
+      logger.error("Non matching data sources on legend item renderers")
+    # Validate label in data_source
+    field_validation = @_check_field_label_on_data_source()
+    if not field_validation
+      logger.error("Bad column name on label: #{@label}")
+
+
+  @define {
+      label: [ p.StringSpec, null ]
+      renderers: [ p.Array, [] ]
+  }
+
+  get_field_from_label_prop: () =>
+    if @label? and @label.field?
+      return @label.field
+
+  get_labels_list_from_label_prop: () =>
+    # Always return a list of the labels
+    if @label? and @label.value?
+      return [@label.value]
+    field = @get_field_from_label_prop()
+    if field?
+      if @renderers[0] and @renderers[0].data_source?
+        source = @renderers[0].data_source
+      else
+        return ["No source found"]
+      if source.get_column?
```

Comment:
This pretty much applies to all uses of `?` operator. If you can avoid it and take advantage of the object model, do it. It states the intent better and is easier on future changes. Also, if you can differentiate between `null` and `undefined`, do it as well (so use e.g. `!= null` instead of `?`). A lot of people in JS community blend those two.